File: app/src/main/java/com/wojdor/hearthstonecards/data/service/mapper/CardMapper.java
package com.wojdor.hearthstonecards.data.service.mapper;

import com.wojdor.hearthstonecards.data.service.model.CardModel;
import com.wojdor.hearthstonecards.domain.Card;

import java.util.ArrayList;
import java.util.List;

public class CardMapper {

    private CardMapper() {
    }

    public static List<Card> map(List<CardModel> models) {
        List<Card> cards = new ArrayList<>();
        for (CardModel model : models) {
            cards.add(map(model));
        }
        return cards;
    }

    public static Card map(CardModel model) {
        Card card = new Card();
        card.setCardId(model.getCardId());
        card.setDbfId(model.getDbfId());
        card.setName(model.getName());
        card.setSet(model.getCardSet());
        card.setRarity(model.getRarity());
        card.setCost(model.getCost());
        card.setText(model.getText());
        card.setFlavorText(model.getFlavor());
        card.setArtist(model.getArtist());
        card.setClassName(model.getPlayerClass());
        return card;
    }
}

File: app/src/main/java/com/wojdor/hearthstonecards/data/service/mapper/CardMapper.kt
package com.wojdor.hearthstonecards.data.service.mapper

import com.wojdor.hearthstonecards.app.extension.empty
import com.wojdor.hearthstonecards.data.service.model.CardModel
import com.wojdor.hearthstonecards.domain.Card
import timber.log.Timber
import java.util.*

object CardMapper {

    fun map(models: List<CardModel>): List<Card> {
        val cards = ArrayList<Card>()
        models.forEach {
            Timber.d(it.cardId)
            cards.add(map(it))
        }
        return cards
    }

    private fun map(model: CardModel) = Card(
            model.dbfId,
            model.cardId,
            model.name,
            model.cardSet,
            model.type,
            model.rarity,
            model.cost,
            model.text ?: String.empty,
            model.flavor ?: String.empty,
            model.artist,
            model.playerClass
    )
}
